src/Descriptor/AssemblyScanner.cs

```diff
@@ -26,6 +29,12 @@ public static AssemblyScanner FindDescriptorsInAssemblyContaining<T>()
             return FindDescriptorsInAssembly(typeof(T).Assembly);
         }
 
+        public AssemblyScanner SetActivator(Func<Type, IDescriptor<IDescriptorContainer>> activator)
+        {
+            DescriptorActivator = activator;
```

Comment:
Since this is a public signature, it would be a good idea to throw if `activator` is `null`.